jwt-auth-common/src/main/java/io/telicent/smart/caches/configuration/auth/RemoteUserInfoLookup.java

```diff
@@ -26,40 +27,61 @@
 import java.time.Duration;
 import java.util.Objects;
 
+/**
+ * A {@link UserInfoLookup} which obtains user info from a remote endpoint
+ */
+@ToString(onlyExplicitlyIncluded = true)
 public class RemoteUserInfoLookup implements UserInfoLookup {
     private final HttpClient http;
     private final ObjectMapper objectMapper;
+    @ToString.Include
     private final Duration timeout;
+    @ToString.Include
+    private final URI endpointUrl;
 
-    public RemoteUserInfoLookup() {
-        this(HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(5)).build(),
-                new ObjectMapper(),
-                Duration.ofSeconds(10));
+    /**
+     * Creates a new remote user info lookup
+     *
+     * @param userInfoEndpoint User Info Endpoint URL
+     */
+    public RemoteUserInfoLookup(String userInfoEndpoint) {
+        this(userInfoEndpoint, HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(5)).build(),
+             new ObjectMapper(), Duration.ofSeconds(10));
     }
 
-    public RemoteUserInfoLookup(HttpClient httpClient, ObjectMapper objectMapper, Duration timeout) {
+    /**
+     * Creates a new customised remote user info lookup
+     *
+     * @param userInfoEndpoint User Info Endpoint URL
+     * @param httpClient       HTTP Client
+     * @param objectMapper     Object mapper for parsing the User Info responses
+     * @param timeout          Timeout for retrieving user info
+     */
+    public RemoteUserInfoLookup(String userInfoEndpoint, HttpClient httpClient, ObjectMapper objectMapper,
+                                Duration timeout) {
+        if (StringUtils.isBlank(userInfoEndpoint)) {
+            throw new IllegalArgumentException("userInfoEndpoint cannot be blank");
+        }
+        this.endpointUrl = URI.create(userInfoEndpoint);
         this.http = Objects.requireNonNull(httpClient);
         this.objectMapper = Objects.requireNonNull(objectMapper);
         this.timeout = timeout != null ? timeout : Duration.ofSeconds(10);
     }
 
     @Override
-    public UserInfo lookup(String userInfoEndpoint, String bearerToken) throws UserInfoLookupException {
-        if (userInfoEndpoint == null || userInfoEndpoint.isBlank()) {
-            throw new UserInfoLookupException("userInfoEndpoint must be provided");
-        }
+    public UserInfo lookup(String bearerToken) throws UserInfoLookupException {
```

Comment:
In actual usage, i.e. not tests, the endpoint URL should be constant so moved to a constructor parameter and refactored accordingly